### ISSUE TITLE
Manual Viewer?

### ISSUE THREAD

**Issue by justingain** [jasarien/Provenance#504]
Don't know if this is a possibility but you don't know unless you ask. Is there a way to incorporate a PDF viewer so that you can view a PDF manual from the menu? I don't see this on emulator front ends and I think it'd be neat to add.

**Comment by jasarien** [comment]
It's a possibility, but as with most things, the difficult bit isn't in adding that functionality, but finding a free-to-use database containing all the required manuals in PDF format and correctly mapping them to ROMs. I don't know of any such database. 

Of course you could always just allow users to import any PDF they like as a manual for a given ROM. Not as sleek then, though.

**Comment by justingain** [comment]
I'd be satisfied with adding my own manuals. I'm not really concerned about having it auto loaded.

**Comment by lolkthxbai** [comment]
When you say manual, you mean the manual that comes with the game?

**Comment by imseandavis** [comment]
http://www.gamesdatabase.org/

This is one a while back a friend of mine wrote some stuff for, it does all the cover art as well as PDF downloads of the manuals for a surprising amount of the games up to current stuff.

It's all pretty easy parsing through parameterized queries, so would be easy to make an API overlay or direct code calls.

Just know that what you think as a pdf for a game may not be what you get as manuals were made in pdf formats back in NES days, it was all handmade print media in corel, lol.

**Comment by jasarien** [comment]
That looks pretty freaking sweet.

**Comment by justingain** [comment]
Yeah that database is pretty awesome. In regards to the Corel deal - that's why I feel it's good to allow users to manually add their own instruction manual.

**Comment by imseandavis** [comment]
So what I meant by corel is that back in the day they didn't have manuals in pdf so at best the NES/SNES/ etc manuals would be page scans into pdf vs native pdf formats, so they aren't as clean as current manuals. They are all PDF docs, so they would be universal, but I agree based on the way Provenance is setup now it would be better to add a manual with the additional code like has been done as with album artwork patch. that would make this a quick patch to add this functionality.

**Comment by imseandavis** [comment]
Found a better one without the past limitations and easier to develop and support

http://replacementdocs.com/download.php

They also have a naming standard making it much easier to retrieve files.

**Comment by sevdestruct** [comment]
wonder if any of these can get us more of the missing metadata per games @JoeMatt and i are looking into for new UI/UX.

**Comment by imseandavis** [comment]
The site listed was just for game manuals in PDF. However, I have a few sites I use for things like arcade machines that would allow metadata queries.

I've used this one in the past: http://wiki.thegamesdb.net/index.php/API_Introduction

There are a few more but some of them are paid.

If you can tell me what metadata you want I'll dig through some old code and see if any of them already point to something that can fill that need.

**Comment by BaffleBlend** [comment]
If I can interject, the PC frontend LaunchBox uses integration with a database called EmuMovies that contains things such as manuals and trailers.

The drawback, unfortunately, is that an account costs money.

**Comment by sevdestruct** [comment]
The upcoming 1.5 release @Joe has implemented links to gamefaqs pages per game which might satiate this some.

**Comment by JoeMatt** [comment]
@claude this could be added to the game more info UI which already shows front and back covers,

we also have the online artwork search ssytem which can search for and add manual artwork files, but our support beyond that is very limited.

for this to be really good, we'd need to piggy back on that online search system for manuals from PVLookup, and support on demand automatic downlaoding and optional to save manual artwork from existing online searches (Any any new sources that specialize in manuals).

manuals should probably also auto show up in the pause menu of the emulator if one exists, whether or not it's locally downloaded

the thought here is that manuals are needed when they're needed, unlike cover artwork which makes sense to add at import for all items to make the UI nice, manuals for more peoeple are an on-demand thing to bother with (though supporting downloading manuals at import time could be an extra feature too, on top of the import process working the same as the batch artwork import or online search features for artwork, which work with more sources than the legacy openvgdb only entries an URls that PVGame is about (the app used to only have openvgdb so some of the artwork and "database id" fields are tied to that outdated and limited databse now)

we never improved the on-import time artwork lookup tow work with more sources than just openvgdb + gamesfaq links, we have all the coded needed, we just need to break out the artwork matching part of import into something more robust for fuzzy searching and online sources)

make an epic, or epics, with the above research and work breakdown as you usually do , and look for any related issues to cross tag and coordinate with
